virttest/qemu_vm.py

```diff
@@ -4804,6 +4810,9 @@ def get_block(self, p_dict={}):
 
         :return: Matched block device name, None when not find any device.
         """
+        if not p_dict:
+            return
```

Comment:
or we could add this part into `def get_block_old(self, blocks_info, p_dict={}):`, for instance:
```
def get_block_old(self, blocks_info, p_dict={}):
...
    if not p_dict:
        return None
    if isinstance(blocks_info, six.string_types):
```
this will be associated with the following codes which to avoid `match=Ture` if `the p_dict={}`.

Reply:
Putting the check in the get_block function is better, the get_block is the final exposed API. 
It avoid redundant calling "info block" .
or add the check in get_block and get_block_old both.